### PR TITLE
[script] [common-arcana] new methods to check currently preparing spell

### PR DESCRIPTION
**Pull request by KatoakDR**
### Background
* I want easier ways to check if you're currently preparing a spell and what spell that is and whether you're likely ready to cast
* Building blocks to greater awesomeness

### Changes
* Add three new methods:
  - `DRCA.spell_preparing?` _are you preparing a spell?_
  - `DRCA.spell_prepared?` _are you prepared to cast the spell you're preparing, i.e. no more cast wait time_
  - `DRCA.spell_preparing` _returns name of spell currently being prepared_

### Tests
_when not preparing a spell_
```
> ,e echo "spell=#{DRCA.spell_preparing}, prepared=#{DRCA.spell_prepared?}, preparing=#{DRCA.spell_preparing?}, waitcastrt=#{checkcastrt}"

[exec1: spell=, prepared=false, preparing=false, waitcastrt=0]
```
_when preparing a spell_
```
> prepare cv 10
You raise your palms skyward, chanting the equation of the Clear Vision spell.

> ,e echo "spell=#{DRCA.spell_preparing}, prepared=#{DRCA.spell_prepared?}, preparing=#{DRCA.spell_preparing?}, waitcastrt=#{checkcastrt}"

[exec1: spell=Clear Vision, prepared=false, preparing=true, waitcastrt=15.854858016967775]
```
_when spell is ready to cast_
```
You feel fully prepared to cast your spell.

> ,e echo "spell=#{DRCA.spell_preparing}, prepared=#{DRCA.spell_prepared?}, preparing=#{DRCA.spell_preparing?}, waitcastrt=#{checkcastrt}"

[exec1: spell=Clear Vision, prepared=true, preparing=true, waitcastrt=0]
```
_after casting spell_
```
> cast
You gesture.
You briefly feel less aware of your environment, then feel more aware again.

> ,e echo "spell=#{DRCA.spell_preparing}, prepared=#{DRCA.spell_prepared?}, preparing=#{DRCA.spell_preparing?}, waitcastrt=#{checkcastrt}"

[exec1: spell=, prepared=false, preparing=false, waitcastrt=0]
```

### CODE REVIEW

File: common-arcana.lic [hunk 1]
```diff
@@ -166,6 +166,26 @@ module DRCA
     match
   end
 
+  # Returns true if preparing a spell, false otherwise.
+  # For the name of the spell, see `get_preparing_spell`
```
Comment (rpherbig) [inline common-arcana.lic:170]: Should this reference `spell_preparing`?

Comment (KatoakDR) [reply common-arcana.lic:170]: Sure, I'll need to tweak both methods a bit to avoid the circular reference when I do that

Comment (rpherbig) [reply common-arcana.lic:170]: I only meant the comment, but if you spotted some other refactoring you want to do, sure thing

Comment (KatoakDR) [reply common-arcana.lic]: Yep, saw an opportunity to reduce number of times I referenced XMLData.prepared_spell, thanks

**Pull request merged**